### PR TITLE
Ignore download links and add router-ignore attribute

### PR DESCRIPTION
**Pull request by benstevens48**
This makes a small change to the DefaultLinkHandler (src/link-handler). Ignores links with a download attribute. Also ignores links with an attribute called rotuer-ignore. Not sure if this is the best name. In addition I have removed the check (targetWindow === 'top' && win === win.top);. this is because it was clearly meant to be _top not top. I could correct it but I don't want to break any sites which might have been using target="_top" as a workaround. I can put it back in if desired, but I can't see why you would use target="_top" if you wanted the router to handle the link. (Also if this is included then there ought to be a similar check for _parent.)

PS any advice on how to avoid updating the dist folder in the pull request without updating .gitignore would be useful - my GIT knowledge is not great. Or can we add dist to .gitignore?

### CODE REVIEW

File: src/link-handler.js [hunk 1]
```diff
@@ -92,6 +92,10 @@ export class DefaultLinkHandler extends LinkHandler {
       return info;
     }
 
+    if (target.hasAttribute('download') || target.hasAttribute('router-ignore')) {
```
Comment (davismj) [inline src/link-handler.js:95]: I'll need to think about a "router-ignore" attribute.

File: src/link-handler.js [hunk 2]
```diff
@@ -135,7 +139,6 @@ export class DefaultLinkHandler extends LinkHandler {
 
     return !targetWindow ||
       targetWindow === win.name ||
-      targetWindow === '_self' ||
-      (targetWindow === 'top' && win === win.top);
```
Comment (davismj) [inline src/link-handler.js:139]: Why did you delete this line of code?

**Review by davismj — CHANGES_REQUESTED** [review]
I'm not so sure about adding an attribute. I wanna think about it just a little bit before we go down that path, but it might be a nice solution.

Also, adding unit tests and documentation for this feature would be a big help. Testing might be a little daunting as I don't see any similar tests, but a short documentation should be straightforward. I'll make sure it gets into the right place when the time comes.

File: src/link-handler.js (hunk 2 above)
Comment (benstevens48) [reply src/link-handler.js:139]: I mentioned this in my comment. I removed `(targetWindow === 'top' && win === win.top)` because it should have been `targetWindow === '_top'` and I didn't want to fix it in case anyone is using `target="_top"` as a workaround for this problem.

File: src/link-handler.js (hunk 1 above)
Comment (benstevens48) [reply src/link-handler.js:95]: Ok, my other possible solution was to just ignore any links with a target attribute, but I thought you might not like that.

**Comment by benstevens48** [comment]
Where should the documentation go? In the comment on the `DefaultLinkHandler` class?

File: src/link-handler.js (hunk 2 above)
Comment (davismj) [reply src/link-handler.js:139]: I don't follow, though. It should be _top, but that wouldn't work because this code is wrong. So maybe someone is using target="top" instead right? And then this would break that?

**Comment by davismj** [comment]
It should probably be in the router documentation. If you can put a quick markdown doc with an explanation of this feature,, I'll get it in the right place. You could create a gist and link it in the PR, that would be great.

File: src/link-handler.js (hunk 2 above)
Comment (benstevens48) [reply src/link-handler.js:139]: No, sorry, I meant that perhaps someone is using target="_top" to prevent the router from processing the link, and fixing the code would break that. For example the original poster of <https://github.com/aurelia/router/issues/428> did this. But it's completely up to you. If you do include that line, I suggest for completeness sake including another line with parent replacing top.

**Comment by benstevens48** [comment]
Ok, here is a gist <https://gist.github.com/benstevens48/9455d4bd309ac3b38007483c1a38a3e1>.

**Comment by Alexander-Taran** [comment]
Love `router-ignore`.. clear purpose
